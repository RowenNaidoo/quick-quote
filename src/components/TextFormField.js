import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const TextFormField = ({ value, fieldId, headerText, onFieldChanged, errorMessage, isRequired, leftRadius = true }) => (
  <Container>
    <Element>
      {headerText}{isRequired && <RequiredIndicator>*</RequiredIndicator>}
    </Element>
    <Element>
      <Field
        type='text'
        id={fieldId}
        value={value}
        onChange={(e) => { onFieldChanged(fieldId, e.target.value) }}>
      </Field>
    </Element>
    <Element>
      <ValidationMessage>
        {errorMessage}
      </ValidationMessage>
    </Element>
  </Container>
)

const Container = styled.div`
  display: flex;
  flex-direction: column;
`
const Element = styled.div`
  flex: 1 1 auto;
  justify-content: left;
  display: flex;
`
const Field = styled.input`
  height: 34px;
  width: 100%;
  padding: 0 10px;
  border-radius: 3px;
  border: 1px solid lightgray;
  flex: 1;
  border-top-left-radius: ${props => props.leftRadius || '0px'};
  border-bottom-left-radius: ${props => props.leftRadius || '0px'};
`
const ValidationMessage = styled.span`
  color: red;
  font-size: 14px;
`
const RequiredIndicator = styled.span`
  color: red;
  font-size: 20px;
`

TextFormField.propTypes = {
  value: PropTypes.string,
  fieldId: PropTypes.string,
  headerText: PropTypes.string,
  onFieldChanged: PropTypes.func,
  errorMessage: PropTypes.string,
  isRequired: PropTypes.bool,
  leftRadius: PropTypes.bool
}

export default TextFormField