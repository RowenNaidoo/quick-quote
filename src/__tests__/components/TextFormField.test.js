import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import TextFormField from '../../components/TextFormField'


describe('<TextFormField />', () => {
  let component
  let mockFieldChanged = jest.fn()

  const defaultProps = {
    value: 'some text',
    fieldId: 'fieldId',
    headerText: 'some header text',
    onFieldChanged: mockFieldChanged,
    errorMessage: '',
    isRequired: true
  }

  beforeEach(() => {
    component = shallow(<TextFormField {...defaultProps} />)
  })

  test('matches snapshot', () => {
    const component = renderer.create(<TextFormField {...defaultProps} />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('fires onChange function', () => {
    const event = {
      target: {value:'a'}
    }

    component.find('#fieldId').simulate('change', event)
    expect(mockFieldChanged).toHaveBeenCalled()
  })
})