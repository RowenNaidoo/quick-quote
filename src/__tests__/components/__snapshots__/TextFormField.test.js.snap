// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextFormField /> matches snapshot 1`] = `
<div
  className="sc-AxjAm iPShpD"
>
  <div
    className="sc-AxirZ bqmkbb"
  >
    some header text
    <span
      className="sc-AxhUy enGJvm"
    >
      *
    </span>
  </div>
  <div
    className="sc-AxirZ bqmkbb"
  >
    <input
      className="sc-AxiKw jgecmR"
      id="fieldId"
      onChange={[Function]}
      type="text"
      value="some text"
    />
  </div>
  <div
    className="sc-AxirZ bqmkbb"
  >
    <span
      className="sc-AxhCb jNwDnw"
    >
      
    </span>
  </div>
</div>
`;
